### ISSUE TITLE
Pastel Failure on windows terminal that doesn't support colors.

### ISSUE THREAD

**Issue by jordwalke** [facebookexperimental/reason-native#111]
Here's a [failure](https://dev.azure.com/pesy/pesy/_build/results?buildId=28&view=logs&jobId=2d2b3007-3c5c-5840-9bb0-2b1ea49925f3&taskId=be3c55bf-a612-5e2a-92ac-aae36da99334&lineStart=57&lineEnd=58&colStart=1&colEnd=1) log for testing the upcoming `pesy` project: 

@saitonakamura Are you aware of what's going on here?

**Comment by jordwalke** [comment]
It says the error is getting thrown in the C stub but it doesn't give much information about where in that C stub.

cc @prometheansacrifice who discovered this bug.

**Comment by jordwalke** [comment]
Here's the specific output:

```
Fatal error: exception (Failure "in enable_windows_console_ansi_seqences: ")
Raised by primitive operation at file "src/pastel/WinConsoleColorsSupport.re", line 10, characters 17-58
Called from file "src/pastel/TerminalImplementation.re", line 10, characters 0-32
##[error]Cmd.exe exited with code '-1'.
Test failed: Non zero exit when running bootstrapper
##[section]Finishing: Running tests

```

**Comment by prometheansacrifice** [comment]
Does it really have to fail hard? If it could fail softly it would be really nice.

**Comment by prometheansacrifice** [comment]
Alternatively, something like `.supportsANSIColors()` would be helpful? Better still, Pastel itself would return the strings without the color modifiers incase the terminal doesn't support colors.

**Comment by jordwalke** [comment]
This comment explains perhaps:

https://stackoverflow.com/questions/38772468/setconsolemode-and-enable-virtual-terminal-processing#comment65200767_38869870

The docs don't explicitly say it, but this API with `ENABLE_VIRTUAL_TERMINAL_PROCESSING` is only supported on certain versions of windows. I wonder if the CI is not running something modern enough.  The commenter claims we need `10586` or greater. The docs don't say this though.

**Comment by jordwalke** [comment]
I wonder if `ENABLE_VIRTUAL_TERMINAL_PROCESSING` not being defined is a good indicator that we can't pass it to `SetConsoleMode` successfully, and in that case, just avoid passing it.

What version of Windows were @prometheansacrifice's CI running on?

**Comment by prometheansacrifice** [comment]
vmImage: vs2017-win2016
As per the docs thats Visual Studio 2017 on Windows Server 2016 
[Link](https://docs.microsoft.com/en-us/azure/devops/pipelines/agents/hosted?view=azure-devops&tabs=yaml#use-a-microsoft-hosted-agent)

**Comment by jordwalke** [comment]
Looks like it's not new enough (I think). See build `10586` here: https://en.wikipedia.org/wiki/Windows_10_version_history

**Comment by jordwalke** [comment]
Found microsoft's description of what "Hosted VS2017" means: https://github.com/Microsoft/azure-pipelines-image-generation/blob/master/images/win/Vs2017-Server2016-Readme.md

**Comment by jordwalke** [comment]
More stack overflow describing how incorrectly documented that flag is: https://stackoverflow.com/questions/46030331/enable-virtual-terminal-processing-and-disable-newline-auto-return-failing

**Comment by jordwalke** [comment]
Turns out I can repro this locally. I think it's because I'm running in a mingw shell and so does Azure by default. We must have configured to not run in mingw environment. We need to make sure this works in mingw. Working on a fix.

**Comment by jordwalke** [comment]
Findings: In a mingw environment the `GetConsoleMode` isn't even returning successfully. `GetLastError()` returns `6`.

**Comment by jordwalke** [comment]
Fix: https://github.com/facebookexperimental/reason-native/pull/112/files

**Comment by bandersongit** [comment]
Included in Pastel 0.2.1
